viewing_party/party.py

```diff
@@ -1,23 +1,164 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        return {"title": title, "genre": genre, "rating": rating}
+    return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total = 0
+
+    for movie in user_data["watched"]:
+        total += movie["rating"]
+
+    if len(user_data["watched"]) != 0:
+        total /= len(user_data['watched'])

```

Comment:
On `line 34` you are essentially checking if the `watched` list on a user object is empty (a guard clause). I would suggest to change like the following: 
```suggestion
    if len(user_data["watched"]) == 0:
        return total
        
    for movie in user_data["watched"]:
        total += movie["rating"]

    total /= len(user_data['watched'])
```
If we are checking to see if the list is empty or not then it makes more sense to check before we do any operations over it. If it does turn out to be empty then we can return `total` (being `0` at the is point in our logic) since there is no average or anything to find.